ModelImporter.cpp

```diff
@@ -85,12 +85,23 @@ Status importInput(ImporterContext* importer_ctx,
 
 Status importInputs(ImporterContext* importer_ctx,
                     ::ONNX_NAMESPACE::GraphProto const& graph,
-                    string_map<TensorOrWeights>* tensors) {
+                    string_map<TensorOrWeights>* tensors,
+                    uint32_t weights_count,
+                    onnxTensorDescriptor const* weight_descriptors) {
+  // The weights may come from two sources:
+  // 1. Initializer list in onnx graph
+  // 2. User specified weight through onnxifi
   string_map<::ONNX_NAMESPACE::TensorProto const*> initializer_map;
   for( ::ONNX_NAMESPACE::TensorProto const& initializer : graph.initializer() ) {
     ASSERT(!initializer_map.count(initializer.name()), ErrorCode::kINVALID_GRAPH);
     initializer_map.insert({initializer.name(), &initializer});
   }
+  string_map<onnxTensorDescriptor const*> weight_map;
+  for (uint32_t i = 0; i < weights_count; ++i) {
+    onnxTensorDescriptor const* desc = weight_descriptors + i;
+    ASSERT(desc, ErrorCode::kINVALID_VALUE);
```

Comment:
This check is useful only when `i == 0` when it de-factor validates `weight_descriptors != NULL`. Would be better to do `weight_descriptors != NULL` check directly.